internal/server/v2/graph.go

```diff
@@ -25,8 +25,8 @@ type Arc struct {
 	Decorator string
 	// The properties of the arc. This is when property is specified with []
 	BracketProps []string
-	// The filter of the arc.
-	Filter map[string]string
+	// The filter of the arc: filter key -> filter value set.
+	Filter map[string]map[string]struct{}
```

Comment:
wonder why use set instead of just array? Then all the uses of this filter looks simpler, and only need to dedupe it once in the building (and in most of the cases, it only has one element).

Reply:
done